src/eos-updater-data.c

```diff
@@ -22,17 +22,31 @@
 
 #include "eos-updater-data.h"
 
+#include "eos-util.h"
+
 #include <string.h>
 
-void
+gboolean
 eos_updater_data_init (EosUpdaterData *data,
-                       OstreeRepo *repo)
+                       OstreeRepo *repo,
+                       GError **error)
 {
-  g_return_if_fail (data == NULL);
-  g_return_if_fail (OSTREE_IS_REPO (repo));
+  if (data == NULL)
+    return TRUE;
```

Comment:
I looked a bit at the prior art - `g_variant_builder_{init,clear}` do no checks for the `NULL` pointer at all.

Alright then, programmer's error it is.